packages/components/upload/src/upload-content.vue

```diff
@@ -92,9 +92,14 @@ const upload = async (rawFile: UploadRawFile) => {
   let beforeData: UploadContentProps['data'] = {}
 
   try {
+    // origin data: Handle data changes after synchronization tasks are executed
+    const originData = props.data
     const beforeUploadPromise = props.beforeUpload(rawFile)
     beforeData = isObject(props.data) ? cloneDeep(props.data) : props.data
     hookResult = await beforeUploadPromise
+    if (isObject(props.data) && isEqual(originData, beforeData)) {
```

Comment:
i think this line should be changed to ` if (isObject(props.data) && !isEqual(originData, beforeData))`. Because we only need to change ‘beforeData’ when props.data is updated after Promise task is resolved.

Reply:
At this time, `beforeData` is still the value under the synchronization task.

Reply:
![image](https://user-images.githubusercontent.com/23251408/235042491-bd56d73a-70cc-4079-b4fa-8cb676267c56.png)

Reply:
@chenxch , i add some test code to explain it more clearly in https://github.com/element-plus/element-plus/pull/12655

Reply:
@Benleie Good job, I like your proof, but I think that's a wrong usage, and you shouldn't construct an asynchronous processing inside a promise. If you judge according to your judgment, then there is a problem with the more common beforeupload below.

```ts
const beforeUpload: UploadProps['beforeUpload'] = (rawFile: UploadRawFile) => {
  return new Promise(resolve => {
      uploadData.key = `${rawFile.name}`;
      resolve(rawFile)
  });
};
```
At this time, all requested keys will become the same value.

Reply:
okay！